arch/xtensa/core/userspace.S

```diff
@@ -255,6 +292,7 @@ xtensa_userspace_enter:
 	l32i a6, a1, 24
 	call4 xtensa_swap_update_page_tables
 
+#if XCHAL_HAVE_THREADPTR
```

Comment:
Isn't `CONFIG_THREAD_LOCAL_STORAGE` checking for this condition ? I mean, there is no TLS without THREADPTR.

Reply:
This is needed as the `#else` part inside `#ifdef CONFIG_THREAD_LOCAL_STORAGE` still needs `THREADPTR`.